pymbar/tests/test_covariance.py

```diff
@@ -33,7 +35,8 @@ def _test(data_generator):
     eq(pymbar.mbar_solvers.self_consistent_update(U, N_k, mbar.f_k), mbar.f_k, decimal=10)
 
     # Test against old MBAR code.
-    mbar0 = pymbar.old_mbar.MBAR(U, N_k)
+    with suppress_warnings():
```

Comment:
Why is suppress_warnings needed here?

Reply:
It suppresses some [very specific warnings](https://github.com/choderalab/pymbar/blob/fe5731312db92a7a396b777ee455c1ec382d4c1e/pymbar/utils_for_testing.py#L66-L68) complaining about methods which don't have a Hessian or a Jacobian, but will still solve the correct way. This only comes up in the test suite and if a user ever uses those methods, the warnings print as normal. I could change the function to be called something like `suppress_derivative_warnings_for_test` to be more explicit if ti helps?

Reply:
Ah, OK. that makes sense. Yeah, probably better to be more explicit in the name.